java/test/jmri/implementation/AbstractReporterTestBase.java

```diff
@@ -78,6 +78,14 @@ public void testPropertyChange() {
         // Check that LastReport was not seen (no change on null)
         Assert.assertFalse("LastReport seen after null", lastReportSeen);
     }
+    
+    @Test
+    public void testDispose() {
+        r.addPropertyChangeListener(new TestReporterListener());
+        Assert.assertEquals("controller listener added", 1, r.getNumPropertyChangeListeners());
+        r.dispose();
+        Assert.assertEquals("controller listeners remaining", 0, r.getNumPropertyChangeListeners());
```

Comment:
The expected behavior of [`NamedBean.dispose()`](https://www.jmri.org/JavaDoc/doc/jmri/NamedBean.html#dispose()) is that the NamedBean is no longer valid after the dispose call.

What this means is that when the call to get the number of property change listeners is made after the call to dispose a numeric result is not the only valid outcome. 

An exception may occur, and As a developer, I would anticipate this occurring based on the Javadoc.

Reply:
Have modified the test to comply with the NamedBean.getNumPropertyChangeListeners Javadoc, "Number of current listeners. May return -1 if the information is not available for some reason."